phpBB/includes/functions.php

```diff
@@ -1661,7 +1661,8 @@ function generate_board_url($without_script_path = false)
 	}
 	else
 	{
-		$server_port = $request->server('SERVER_PORT', 0);
+		$server_port = $symfony_request->getPort();
```

Comment:
Symfony request does not cast to integer. Please make sure this is casted to int.